include/framework/i_observer.h

```diff
@@ -1,13 +1,23 @@
 #ifndef ACID_INCLUDE_I_OBSERVER_H_
 #define ACID_INCLUDE_I_OBSERVER_H_
 
+template <class T>
+class IObservable;
+
 template <class T>
 class IObserver {
   public:
-	virtual ~IObserver() = default;
+	virtual ~IObserver() {
+		if (observable != nullptr) {
+			observable->delete_observer(this);
```

Comment:
Observers are Systems, Nodes and Proxies. Node dies only when it's entity dies, so we really don't have to invoke `delete_observer()`